src/curl.cpp

```diff
@@ -4232,6 +4232,10 @@ bool S3fsCurl::UploadMultipartPostComplete()
         }
     }
     partdata.petag->etag = it->second;
+    if(partdata.petag->etag.size() >= 2 && *partdata.petag->etag.begin() == '"' && *partdata.petag->etag.rbegin() == '"'){
+        partdata.petag->etag.erase(partdata.petag->etag.size() - 1);
+        partdata.petag->etag.erase(0, 1);
+    }
```

Comment:
Could you abstract the etag erasing logic from `S3fsCurl::CopyMultipartPostComplete` and `etag_equals` into a helper function?

Reply:
Changed to use `trim` string function(defined in `string_util.h`).
Strictly speaking, it no longer only removes pairs of double quotes, but I used this on the assumption that the `ETag` response would not contain anything like that.
The same thing can be done by implementing it in the `ETag` key acquisition method, but it seems to be complicated, so this modification method is simpler.
I have rebased it, so please review it again.